Sources/ScrollingContentView.swift

```diff
@@ -105,51 +105,51 @@ public class VerticalScrollingContentView: UIView {
-    let leadingAttribute: NSLayoutAttribute = self.dynamicType.scrollDirection == .Vertical ? .Leading : .Top
-    let trailingAttribute: NSLayoutAttribute = self.dynamicType.scrollDirection == .Vertical ? .Trailing : .Bottom
+    let topAttribute: NSLayoutAttribute = type(of: self).scrollDirection == .vertical ? .top : .leading
+    let bottomAttribute: NSLayoutAttribute = type(of: self).scrollDirection == .vertical ? .bottom : .trailing
+    let leadingAttribute: NSLayoutAttribute = type(of: self).scrollDirection == .vertical ? .leading : .top
+    let trailingAttribute: NSLayoutAttribute = type(of: self).scrollDirection == .vertical ? .trailing : .bottom
     
     // the content view is constrained to the top and bottom of the scroll view
-    addConstraintToView(scrollView, targetAttribute: topAttribute, fromContentViewAttribute: topAttribute)
-    addConstraintToView(scrollView, targetAttribute: bottomAttribute, fromContentViewAttribute: bottomAttribute)
+    addConstraint(toView: scrollView, targetAttribute: topAttribute, fromContentViewAttribute: topAttribute)
+    addConstraint(toView: scrollView, targetAttribute: bottomAttribute, fromContentViewAttribute: bottomAttribute)
     
     // the content view is constrained to the left and right of the view itself (not its scrollview!)
-    addConstraintToView(self, targetAttribute: leadingAttribute, fromContentViewAttribute: leadingAttribute)
-    addConstraintToView(self, targetAttribute: trailingAttribute, fromContentViewAttribute: trailingAttribute)
+    addConstraint(toView: self, targetAttribute: leadingAttribute, fromContentViewAttribute: leadingAttribute)
+    addConstraint(toView: self, targetAttribute: trailingAttribute, fromContentViewAttribute: trailingAttribute)
   }
   
-  private func addConstraintToView(targetView: UIView, targetAttribute: NSLayoutAttribute, fromContentViewAttribute sourceAttribute: NSLayoutAttribute) {
-    let constraint = NSLayoutConstraint(item: contentView, attribute: sourceAttribute, relatedBy: NSLayoutRelation.Equal, toItem: targetView, attribute: targetAttribute, multiplier: 1, constant: 0)
+  private func addConstraint(toView targetView: UIView, targetAttribute: NSLayoutAttribute, fromContentViewAttribute sourceAttribute: NSLayoutAttribute) {
+    let constraint = NSLayoutConstraint(item: contentView, attribute: sourceAttribute, relatedBy: NSLayoutRelation.equal, toItem: targetView, attribute: targetAttribute, multiplier: 1, constant: 0)
     targetView.addConstraint(constraint)
   }
   
   // MARK: Static helpers
   
   private static func newScrollView() -> UIScrollView {
-    let scrollView = UIScrollView(frame: CGRectZero)
-    scrollView.backgroundColor = UIColor.clearColor()
-    scrollView.autoresizingMask = [.FlexibleWidth, .FlexibleHeight]
+    let scrollView = UIScrollView(frame: CGRect.zero)
+    scrollView.backgroundColor = UIColor.clear
```

Comment:
no need to change but just fyi you could use `= .clear` and `frame: .zero` for simplicity